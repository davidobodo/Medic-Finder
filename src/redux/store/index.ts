import {createStore, applyMiddleware} from "redux";
import createSagaMiddleware from "redux-saga";

import rootReducer from "../reducers";
import rootSaga from "../sagas";

const sagaMiddleware = createSagaMiddleware();

//add saga middleware for taking care of making network requests
const store = createStore(rootReducer, applyMiddleware(sagaMiddleware));

sagaMiddleware.run(rootSaga);

export default store;
