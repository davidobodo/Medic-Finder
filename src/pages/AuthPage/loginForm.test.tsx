import React from 'react';
import { Provider } from 'react-redux';
import store from '../../redux/store';

import { cleanup, fireEvent, render } from '@testing-library/react';
import '@testing-library/jest-dom';

import Login from './login/login';

afterEach(cleanup);

const LoginProps = {
	onRequireSignUp: jest.fn()
};

const app = (
	<Provider store={store}>
		<Login {...LoginProps} />
	</Provider>
);

//--------------------------------------------------------------
// Snapshot
//--------------------------------------------------------------

it('should take a snapshot', () => {
	const { container } = render(app);
	expect(container).toMatchSnapshot();
});

//--------------------------------------------------------------
// Successfully changing input field values
//--------------------------------------------------------------

it('Email should change input value', () => {
	const input = render(app).queryByPlaceholderText('email') as HTMLInputElement;
	fireEvent.change(input, { target: { value: 'obodo@enye.com' } });
	expect(input.value).toBe('obodo@enye.com');
});

it('Password should change input value', () => {
	const input = render(app).queryByPlaceholderText('password') as HTMLInputElement;
	fireEvent.change(input, { target: { value: '123456' } });
	expect(input.value).toBe('123456');
});
