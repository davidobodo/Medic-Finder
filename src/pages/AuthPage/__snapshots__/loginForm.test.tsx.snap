// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<div>
  <section
    class="makeStyles-formContainer-10"
  >
    <h4
      class="MuiTypography-root makeStyles-header-11 MuiTypography-h4"
    >
      Login
    </h4>
    <form
      novalidate=""
    >
      <div
        class="makeStyles-inputContainer-13"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
            data-shrink="true"
            for="standard-full-width"
            id="standard-full-width-label"
          >
            Email
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              id="standard-full-width"
              name="email"
              placeholder="email"
              type="email"
              value=""
            />
          </div>
          
        </div>
      </div>
      <div
        class="makeStyles-inputContainer-13"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
            data-shrink="true"
            for="standard-full-width"
            id="standard-full-width-label"
          >
            Password
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              id="standard-full-width"
              name="password"
              placeholder="password"
              type="password"
              value=""
            />
          </div>
          
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained Mui-disabled Mui-disabled"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Login
        </span>
      </button>
    </form>
    <div
      class="makeStyles-enquiry-14"
    >
      Don't have an account? 
      <button>
        Sign Up
      </button>
    </div>
  </section>
</div>
`;
